### PR TITLE
List ops: sync expected test results and input data with problem-specifications.

### PR DESCRIPTION
**Pull request by BethanyG**
Part of https://github.com/exercism/python/issues/1762

- Changed function names to  **map** and **filter** in tests, `example.py`, and exercise stub to conform with `canonical-data.json`.
- Changed changed function argument names to **list**, **inital**, and **function** in `example.py` and exercise stub to better conform to `canonical-data.json`.
- Changed references to single, two-letter, and abbreviated variable names in `example.py` to whole words for clarity.

### CODE REVIEW

File: exercises/list-ops/list_ops_test.py [hunk 1]
```diff
@@ -34,11 +34,11 @@ def test_concat_list_of_nested_lists(self):
 
     # tests for filter_clone
```
Comment (cmccandless) [inline exercises/list-ops/list_ops_test.py]:
```suggestion
    # tests for filter
```

File: exercises/list-ops/list_ops_test.py [hunk 2]
```diff
@@ -50,10 +50,10 @@ def test_length_nonempty_list(self):
 
     # tests for map_clone
```
Comment (cmccandless) [inline exercises/list-ops/list_ops_test.py]:
```suggestion
    # tests for map
```

**Comment by BethanyG** [comment]
Yeah.  I do have to admit that this one made my heart quail, a bit.  But agreed:  the point here is to re-invent the wheel, and hopefully code like this wouldn't see the light of day outside this exercise.

**Comment by cmccandless** [comment]
Merged; thanks for working on this!

**Pull request merged**